packages/techdocs-common/src/stages/publish/helpers.ts

```diff
@@ -82,3 +82,24 @@ export const getFileTreeRecursively = async (
   });
   return fileList;
 };
+
+/**
+ * Returns the version of an object's storage path where the first three parts
+ * of the path (the entity triplet of namespace, kind, and name) are
+ * lower-cased.
+ *
+ * Path must not include a starting slash.
+ *
+ * @example
+ * lowerCaseEntityTripletInStoragePath('default/Component/backstage')
+ * // return default/component/backstage
+ */
+export const lowerCaseEntityTripletInStoragePath = (
+  originalPath: string,
+): string => {
+  const [namespace, kind, name, ...parts] = originalPath.split('/');
```

Comment:
we may need to check if we have a minimum of 3 parts in the part other than the filename.

Reply:
Yeah, good call.  Maybe we throw/catch and log a warning in that case (`Encountered file unmanaged by TechDocs ${fileName}. Skipping.`)

Reply:
Hey @soapraj can you review it again, please?